src/Compilers/CSharp/Portable/Compilation/SyntaxTreeSemanticModel.cs

```diff
@@ -2033,9 +2037,7 @@ internal override ImmutableArray<ISymbol> GetDeclaredSymbols(BaseFieldDeclaratio
                 return null;
             }
 
-            return
-                GetParameterSymbol(method.Parameters, parameter, cancellationToken) ??
-                ((object)method.PartialDefinitionPart == null ? null : GetParameterSymbol(method.PartialDefinitionPart.Parameters, parameter, cancellationToken));
```

Comment:
I believe this check of 'PartialDefinitionPart' did not make sense. The containing method `GetMethodParameterSymbol(ParameterSyntax, CancellationToken)` works by grabbing the parent `MemberDeclarationSyntax` of the given `ParameterSyntax`. It then gets the symbol for the `MemberDeclarationSyntax`, and searches the symbol's parameters for one which matches the original input `ParameterSyntax`.

We should expect that `GetDeclaredSymbol` returns the correct partial symbol corresponding to the declaration syntax which was passed in. So there is no benefit, in the case where searching the parameters of the symbol we got resulted in no match, to also searching the parameters of the definition part which was related to the symbol we got.